src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,62 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+
+  def toBoxes(x: Double, y: Double, xmin: Double, xmax: Double, ymin: Double, ymax: Double, nDivisions: Int): (Int, Int) = {
+    val i = ((x - xmin) / (xmax - xmin) * (if (x.toInt == xmax) nDivisions - 1 else nDivisions)).toInt
```

Comment:
I would write this:

```
  var i = ((x - xmin) / (xmax - xmin)).toInt
  if (i == nDivisions)
    i = nDivisions - 1
  assert(i >= 0 && i < nDivisions)
```

In general, comparing integers is safer than comparing floating point numbers.

Reply:
@cseed Is this supposed to check if `i == nDivisions`? I think for this check to pass, x must be much larger than xmax